### ISSUE TITLE
Assertion failed: ((obj)->tt != MRB_TT_FREE)

### ISSUE THREAD

**Issue by ksss** [mruby/mruby#3542]
mruby: 051e40c0
OS: macOS
compiler: clang v8.0.0

build_config.rb

```rb
MRuby::Build.new do |conf|
  toolchain :gcc
  enable_debug
  conf.gembox 'default'
  cc.defines << %w(MRB_GC_STRESS)
end
```

t.rb

```rb
def foo(*)
end
puts foo('a', 'b', 'c', 'd', 'e')
```

```
$ mruby t.rb
Assertion failed: ((obj)->tt != MRB_TT_FREE), function mrb_gc_mark, file /Users/ksss/src/github.com/ksss/mruby/src/gc.c, line 698.
[2]    30544 abort      mruby t.rb
```

```
$ lldb mruby
(lldb) target create "/Users/ksss/bin/mruby"
Current executable set to '/Users/ksss/bin/mruby' (x86_64).
(lldb) run t.rb
Process 30572 launched: '/Users/ksss/bin/mruby' (x86_64)
Assertion failed: ((obj)->tt != MRB_TT_FREE), function mrb_gc_mark, file /Users/ksss/src/github.com/ksss/mruby/src/gc.c, line 698.
Process 30572 stopped
* thread #1: tid = 0x112a84a, 0x00007fff9f5a4dd6 libsystem_kernel.dylib`__pthread_kill + 10, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00007fff9f5a4dd6 libsystem_kernel.dylib`__pthread_kill + 10
libsystem_kernel.dylib`__pthread_kill:
->  0x7fff9f5a4dd6 <+10>: jae    0x7fff9f5a4de0            ; <+20>
    0x7fff9f5a4dd8 <+12>: movq   %rax, %rdi
    0x7fff9f5a4ddb <+15>: jmp    0x7fff9f59dcdf            ; cerror_nocancel
    0x7fff9f5a4de0 <+20>: retq
(lldb) bt
* thread #1: tid = 0x112a84a, 0x00007fff9f5a4dd6 libsystem_kernel.dylib`__pthread_kill + 10, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
  * frame #0: 0x00007fff9f5a4dd6 libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff9f690787 libsystem_pthread.dylib`pthread_kill + 90
    frame #2: 0x00007fff9f50a420 libsystem_c.dylib`abort + 129
    frame #3: 0x00007fff9f4d1893 libsystem_c.dylib`__assert_rtn + 320
    frame #4: 0x0000000100019854 mruby`mrb_gc_mark(mrb=0x0000000100300360, obj=0x0000000101001bc0) + 132 at gc.c:698
    frame #5: 0x000000010001b688 mruby`gc_mark_children(mrb=0x0000000100300360, gc=0x0000000100300438, obj=0x0000000101001bf0) + 680 at gc.c:664
    frame #6: 0x000000010001b11d mruby`gc_gray_mark(mrb=0x0000000100300360, gc=0x0000000100300438, obj=0x0000000101001bf0) + 45 at gc.c:888
    frame #7: 0x000000010001a9e5 mruby`incremental_marking_phase(mrb=0x0000000100300360, gc=0x0000000100300438, limit=18446744073709551615) + 101 at gc.c:983
    frame #8: 0x000000010001a599 mruby`incremental_gc(mrb=0x0000000100300360, gc=0x0000000100300438, limit=18446744073709551615) + 169 at gc.c:1087
    frame #9: 0x0000000100019907 mruby`incremental_gc_until(mrb=0x0000000100300360, gc=0x0000000100300438, to_state=MRB_GC_STATE_ROOT) + 39 at gc.c:1112
    frame #10: 0x0000000100018518 mruby`mrb_full_gc(mrb=0x0000000100300360) + 152 at gc.c:1212
    frame #11: 0x0000000100019417 mruby`mrb_obj_alloc(mrb=0x0000000100300360, ttype=MRB_TT_ARRAY, cls=0x0000000101009840) + 343 at gc.c:504
    frame #12: 0x000000010000192f mruby`ary_new_capa(mrb=0x0000000100300360, capa=1) + 111 at array.c:30
    frame #13: 0x0000000100001a9f mruby`mrb_ary_new_from_values(mrb=0x0000000100300360, size=1, vals=0x000000010100d020) + 31 at array.c:78
    frame #14: 0x0000000100042f8b mruby`mrb_vm_exec(mrb=0x0000000100300360, proc=0x0000000101006240, pc=0x000000010008dabc) + 13787 at vm.c:1607
    frame #15: 0x000000010003f997 mruby`mrb_vm_run(mrb=0x0000000100300360, proc=0x0000000101001c20, self=mrb_value @ 0x00007fff5fbff270, stack_keep=0) + 135 at vm.c:823
    frame #16: 0x0000000100048554 mruby`mrb_top_run(mrb=0x0000000100300360, proc=0x0000000101001c20, self=mrb_value @ 0x00007fff5fbff2e0, stack_keep=0) + 100 at vm.c:2614
    frame #17: 0x0000000100073d0f mruby`mrb_load_exec(mrb=0x0000000100300360, p=0x0000000102000020, c=0x0000000100400130) + 1183 at parse.y:5760
    frame #18: 0x0000000100073f53 mruby`mrb_load_file_cxt(mrb=0x0000000100300360, f=0x00007fffa82ab0b0, c=0x0000000100400130) + 67 at parse.y:5769
    frame #19: 0x00000001000010f4 mruby`main(argc=2, argv=0x00007fff5fbff700) + 916 at mruby.c:227
    frame #20: 0x00007fff9f476255 libdyld.dylib`start + 1
```

**Comment by matz** [comment]
Probably it's fixed by 5c114c9

**Comment by ksss** [comment]
I confirmed that it has been fixed 🌸
